requirements.txt

```diff
@@ -1,12 +1,19 @@
-# Keep this file in sync with `doc/doc-requirements.txt`!
+# About pycurl:
+# - pycurl is strongly recommended for performance but not strictly required by
+#   BinderHub.
```

Comment:
This is hard to understand. The binderhub package authors (us) decided to make pycurl a required dependency by including it in the list of packages required in `setup.py`. So pycurl isn't optional for the user of binderhub.

The package authors could have decided to not make it a dependency. This is when it was optional. So I would rephrase it like this
```suggestion
# - pycurl is a dependency but not listed here, only in `setup.py`
```